misk-testing/src/main/kotlin/misk/service/ServiceTestingModule.kt

```diff
@@ -0,0 +1,92 @@
+package misk.service
+
+import com.google.common.util.concurrent.Service
+import com.google.inject.Key
+import com.google.inject.Provider
+import misk.DependentService
+import misk.inject.KAbstractModule
+import java.util.concurrent.Executor
+import java.util.concurrent.TimeUnit
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+
+/**
+ * [ServiceTestingModule] provides additional help for testing service, notably allowing existing
+ * services to be bound with additional dependencies specifically required for testing. Typically
+ * used when a [Service] requires an external service (e.g. zookeeper, etcd, vitess, etc)
+ * that is being spun up within the test itself
+ */
+class ServiceTestingModule<T : Service> internal constructor(
+  private val wrappedServiceType: KClass<T>,
+  private val extraDependencies: Set<Key<*>>
+) : KAbstractModule() {
+
+  override fun configure() {
+    // Register a wrapper around the service with the service manager. The wrapper delegates
+    // to the underlying service for all calls except [DependentService.consumedKeys], which
+    // it overrides to extend the consumed service list with the extra dependencies
+    multibind<Service>().toProvider(ServiceWithTestDependenciesProvider(
+        getProvider(wrappedServiceType.java),
+        extraDependencies
+    ))
+
+    // Bind the wrapped service directly so that application code can inject it
+    bind(wrappedServiceType.java)
+  }
+
+  companion object {
+    /** @return A [Module] binding the service with extra dependencies */
+    fun <T : Service> withExtraDependencies(
+      wrappedServiceType: KClass<T>,
+      vararg extraDependencies: Key<*>
+    ): com.google.inject.Module =
```

Comment:
why the fully qualified name?

Reply:
because `Module` is now a class in the `java.lang` package